testing/camunda-process-test-java/src/test/java/io/camunda/process/test/api/DecisionMatchedRulesAssertTest.java

```diff
@@ -0,0 +1,240 @@
+    void hasMatchedRules(final EvaluatedDecision decision, final int[] expectedMatchedRules) {
+      // then
+      CamundaAssert.assertThat(decision).hasMatchedRules(expectedMatchedRules);
+    }
+
+    @Test
+    void hasDetailedOutputIfNoMatchesFound() {
+      // then
+      final EvaluatedDecision decision = evaluatedDecision(decisionResult(singleRule()));
+      Assertions.assertThatThrownBy(() -> assertThat(decision).hasMatchedRules(2))
+          .hasMessage(
+              "Expected EvaluatedDecision [name] to have matched rules [2], but did not. Matches:\n"
+                  + "\t- matched: []\n"
+                  + "\t- missing: [2]\n"
+                  + "\t- unexpected: [1]");
+    }
+
+    @Test
+    void hasDetailedOutputIfPartialMatchesFound() {
+      // then
+      final EvaluatedDecision decision = evaluatedDecision(decisionResult(multiRule()));
+      Assertions.assertThatThrownBy(() -> assertThat(decision).hasMatchedRules(1, 2, 4))
+          .hasMessage(
+              "Expected EvaluatedDecision [name] to have matched rules [1, 2, 4], but did not. Matches:\n"
+                  + "\t- matched: [1, 2]\n"
+                  + "\t- missing: [4]\n"
+                  + "\t- unexpected: [3]");
+    }
+  }
+
+  @Nested
+  class HasNotMatchedRules {
+
+    @ParameterizedTest
+    @MethodSource(
+        "io.camunda.process.test.api.DecisionMatchedRulesAssertTest#positiveUnmatchedRuleScenarios")
+    void hasMatchedRules(final EvaluatedDecision decision, final int[] expectedMatchedRules) {
+      // then
+      CamundaAssert.assertThat(decision).hasNotMatchedRules(expectedMatchedRules);
+    }
+
+    @Test
+    void hasDetailedOutputIfMatched() {
+      // then
+      final EvaluatedDecision decision = evaluatedDecision(decisionResult(singleRule()));
+      Assertions.assertThatThrownBy(() -> assertThat(decision).hasNotMatchedRules(1))
+          .hasMessage(
+              "Expected EvaluatedDecision [name] to not have matched rules [1], but matched [1]");
+    }
+
+    @Test
+    void hasDetailedOutputIfPartiallyMatched() {
+      // then
+      final EvaluatedDecision decision = evaluatedDecision(decisionResult(multiRule()));
+      Assertions.assertThatThrownBy(() -> assertThat(decision).hasNotMatchedRules(1, 2, 4))
+          .hasMessage(
+              "Expected EvaluatedDecision [name] to not have matched rules [1, 2, 4], but matched [1, 2]");
+    }
+  }
+
+  @Nested
+  class EdgeCases {
+    @Test
+    void shouldFailIfNoMatchedRulesGiven() {
+      // then
+      final EvaluatedDecision decision = evaluatedDecision(decisionResult(multiRule()));
+      Assertions.assertThatThrownBy(() -> assertThat(decision).hasMatchedRules())
+          .hasMessage("No matched rules given. Please provide at least one matched rule index.");
+
+      Assertions.assertThatThrownBy(() -> assertThat(decision).hasNotMatchedRules())
+          .hasMessage("No matched rules given. Please provide at least one matched rule index.");
+    }
```

Comment:
:wrench: We should add a new assertion for the case that no rules matched. 

It is a valid case that no rules of a decision table match. The decision output is `null`.